dotCMS/src/main/java/com/dotmarketing/business/BaseWebAssetAPI.java

```diff
@@ -477,46 +476,40 @@ public int deleteOldVersions(final Date olderThan, final String type) throws Dot
 
         //No need to run all the SQL is we don't get any inodes from the condition.
         if(!resultInodes.isEmpty()){
-            String parameterPlaceholders;
-            List<String> inodesToDeleteList = Collections.EMPTY_LIST;
-            List<String> inodesToDelete = Collections.EMPTY_LIST;
+            List<String> inodesToDelete = new ArrayList<>();
+            List<List<String>> inodesToDeleteMatrix = new ArrayList<>();
             int truncateAt = 100;
-            int totalIter;
 
             //Fill inodesToDelete where each inode from the result.
             for (Map<String, Object> row : resultInodes) {
                 inodesToDelete.add(row.get("inode").toString());
             }
-            
+
+            //We want to create lists of 100 inodes.
+            while (inodesToDelete.size() >= truncateAt){
+                inodesToDeleteMatrix.add(new ArrayList<>(inodesToDelete.subList(0,truncateAt)));
+                inodesToDelete.subList(0,truncateAt).clear();
+            }
+            inodesToDeleteMatrix.add(new ArrayList<>(inodesToDelete.subList(0, inodesToDelete.size())));

```

Comment:
Just as an alternative, this can be done using Guava:
List<List<String>> inodesToDeleteMatrix = Lists.partition(inodesToDelete, truncateAt);